tests/components/vesync/test_init.py

```diff
@@ -98,3 +106,213 @@ async def test_async_setup_entry__loads_fans(
     assert hass.data[DOMAIN][VS_FANS] == [fan]
     assert not hass.data[DOMAIN][VS_LIGHTS]
     assert hass.data[DOMAIN][VS_SENSORS] == [fan]
+
+
+async def test_async_process_devices__no_devices(
+    hass: HomeAssistant, manager, caplog: pytest.LogCaptureFixture
+) -> None:
+    """Test when manager with no devices is processed."""
+    manager = MagicMock()
+    with patch.object(
+        hass, "async_add_executor_job", new=AsyncMock()
+    ) as mock_add_executor_job:
+        devices = await _async_process_devices(hass, manager)
```

Comment:
Don't access any internal details of the integration in tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Assert that the integration creates the expected device entries in the device registry instead.